File: 백준/Silver/1874. 스택 수열/스택 수열.py
import sys

n = int(sys.stdin.readline())
boolean = True
stack = []
arr = []
ans = []
index = 0
i = 1
for _ in range(n):
    arr.append(int(sys.stdin.readline()))

while True:
    stack.append(i)
    ans.append("+")
    i += 1
    while True:
        if stack[-1] == arr[index]:
            stack.pop()
            ans.append("-")
            index += 1
        else:
            break
        if len(stack) == 0:
            break
    if i > n:
        break
if len(stack) > 0:
    print("NO")
else:
    for i in ans:
        print(i)
    

File: 백준/Silver/1874. 스택 수열/스택 수열.cc
#include <iostream>
#include <bits/stdc++.h>
using namespace std;

int main() {
	int n;
	cin >> n;
	int cnt = 1;
	stack<int> S;
	string ans;
	while(n--){
		int value;
		cin >> value;
		while(cnt <= value){
			S.push(cnt++);
			ans += "+\n";
		}
		if(S.top() != value){
			cout << "NO\n";
			return 0;
		}
		S.pop();
		ans += "-\n";
	}
	cout << ans;
	return 0;
}